blocks/library/tweet/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Internal dependencies
+ */
+import { registerBlock, query } from '../../api';
+import Sandbox from '../../../components/sandbox';
+import Button from '../../../components/button';
+import Placeholder from '../../../components/placeholder';
+
+const { prop } = query;
+
+registerBlock( 'core/tweet', {
+	title: wp.i18n.__( 'Tweet' ),
+	icon: 'twitter',
+
+	category: 'social',
+
+	attributes: {
+		url: prop( '*', 'innerHTML' ), // our html is just a div with the url in, for WP's oembed to process
+	},
+
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.fetchTweet = this.fetchTweet.bind( this );
+			this.state = {
+				url: this.props.attributes.url,
```

Comment:
>So when the button is clicked (or form submitted) I'd use a reference to the text input to set the url, instead of having it set when the text input changes?

Ah, I think I overlooked that. With that in mind, it can be fine to create a copy.

Here's an archived (some outdated syntax) article which explains more of the context and highlights this exception:

https://github.com/facebook/react/blob/8cac523/docs/tips/10-props-in-getInitialState-as-anti-pattern.md

>However, it's not an anti-pattern if you make it clear that the prop is only seed data for the component's internally-controlled state:

Since we don't have as much control over the naming of the incoming prop, except if we were to create an intermediary component or rename the attribute (both of which don't seem ideal), I think it'd be fine to ignore the `initialUrl` recommendation here.